axelrod/tests/unit/test_match.py

```diff
@@ -79,16 +80,15 @@ def test_example_prob_end(self):
         outcomes
         """
         p1, p2 = axl.Cooperator(), axl.Cooperator()
-        match = axl.Match((p1, p2), prob_end=0.5)
-        expected_lengths = [3, 1, 5]
+        expected_lengths = [2, 1, 1]
         for seed, expected_length in zip(range(3), expected_lengths):
-            axl.seed(seed)
+            match = axl.Match((p1, p2), prob_end=0.5, seed=seed)
```

Comment:
Commentary/thinking out loud: This change points out one of the major "fundamental" [great] changes this PR does right? It's confirming that Matches are now also defined by their seeds (so the test is now forced to create a new match for each seed). Cool.

Reply:
That's right. A seed isn't required if the match is deterministic, which means that it's not noisy and both the players are deterministic. Otherwise a seed is required.